src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigProcessing/Processors/TruststorePropertiesProcessor.cs

```diff
@@ -24,19 +24,75 @@ namespace SonarScanner.MSBuild.PreProcessor.AnalysisConfigProcessing.Processors;
 /// Map property name to another property name and/or value to pass them through the SONAR_SCANNER_OPTS
 /// environment variable to Scanner CLI 5.
 /// </summary>
-public class PropertyAsScannerOptsMappingProcessor(
+public class TruststorePropertiesProcessor(
     ProcessedArgs localSettings,
     IDictionary<string, string> serverProperties,
+    IFileWrapper fileWrapper,
     IOperatingSystemProvider operatingSystemProvider)
     : AnalysisConfigProcessorBase(localSettings, serverProperties)
 {
+    const string JAVAX_NET_SSL_TRUST_STORE = "javax.net.ssl.trustStore";
+    const string JAVAX_NET_SSL_TRUST_STORE_TYPE = "javax.net.ssl.trustStoreType";
+    const string JAVAX_NET_SSL_TRUST_STORE_PASSWORD = "javax.net.ssl.trustStorePassword";
+
     public override void Update(AnalysisConfig config)
     {
-        MapProperty(config, "javax.net.ssl.trustStore", PropertyValueOrDefault(SonarProperties.TruststorePath, LocalSettings.TruststorePath), ConvertToJavaPath, EnsureSurroundedByQuotes);
-        var password = PropertyValueOrDefault(SonarProperties.TruststorePassword, LocalSettings.TruststorePassword);
-        MapProperty(config, "javax.net.ssl.trustStorePassword", password, EnsureSurroundedByQuotes);
+        if (new Uri(LocalSettings.ServerInfo.ServerUrl).Scheme != "https"
```

Comment:
This [constructor throws](https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-9.0#system-uri-ctor(system-string)). Consider using [`TryCreate(url, UriKind.Absolute, out var uri)`](https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-9.0#system-uri-ctor(system-string)) instead.